src/tryGetDebouncedCompletions.ts

```diff
@@ -0,0 +1,61 @@
+import * as vscode from "vscode";
+import { Capability, isCapabilityEnabled } from "./capabilities/capabilities";
+import getInlineCompletionItems from "./getInlineCompletionItems";
+import TabnineInlineCompletionItem from "./inlineSuggestions/tabnineInlineCompletionItem";
+import { sleep, timed } from "./utils/utils";
+
+export default async function tryGetDebouncedCompletions(
```

Comment:
rename to `debounceCompletions` - the `try` is confusing imo

Reply:
also, can this be a generic debounce function? something like
```typescript
function debounce<T>(
  action: () => Promise<T>,
  debounceMs: number,
  cancellation: () => Promise<boolean>
```

or something similar

Reply:
bc of the re-fetching getInlineCompletionItems, I'm not sure how generic it could be